tools/torch_quant/torch_quant/quantizer.py

```diff
@@ -258,26 +234,48 @@ def qat_gm(
             w_ob_ctr=ob_types.w_ob_ctr,
             bias_ob_ctr=ob_types.bias_ob_ctr,
         )
-        ctx.modify_graph([
-            set_qconfig,
-            insert_act_observer,
-            # Generally we do not add fake-quant to bias during qat fine-tuning. If
-            # users want to evaluate the accuracy of a model in specific state, they
-            # should use the model returned by `quantizer.quantize`
-            partial(quantizable_module_to_observed, is_observe_bias=False)])
+        if self.backend == Backend.DISC:
+            ctx.modify_graph([
+                set_qconfig,
+                insert_act_observer,
+                insert_w_observer,
+                # Generally we do not add fake-quant to bias during qat fine-tuning. If
+                # users want to evaluate the accuracy of a model in specific state, they
+                # should use the model returned by `quantizer.quantize`
+                quantizable_module_to_observed])
+        else:
+            ctx.act_ob_ctr = self.act_ob_ctr
+            ctx.w_ob_ctr = self.w_ob_ctr
+            ctx.bias_ob_ctr = self.bias_ob_ctr
+            ctx.modify_graph([set_qconfig, fuse_modules, insert_w_observer])
+
+            ctx.act_ob_ctr = ob_types.act_ob_ctr
+            ctx.w_ob_ctr = ob_types.w_ob_ctr
+            ctx.bias_ob_ctr = ob_types.bias_ob_ctr
+            ctx.modify_graph(
```

Comment:
why modify the graph twice?

Reply:
For reference and fbgemm backends, we didn't insert any `w_ob` or `bias_ob` in the `quantize` stage. Hence we should firstly insert normal `w_ob` or `bias_ob`, then replace to LSQObserver.